storage/series/series.go

```diff
@@ -433,52 +440,61 @@ func (s *dbSeries) addBlockWithLock(b block.DatabaseBlock) {
 // data in memory during bootstrapping. If that becomes a problem, we could
 // bootstrap in batches, e.g., drain and reset the buffer, drain the streams,
 // then repeat, until len(s.pendingBootstrap) is below a given threshold.
-func (s *dbSeries) Bootstrap(blocks block.DatabaseSeriesBlocks) error {
+func (s *dbSeries) Bootstrap(bootstrappedBlocks block.DatabaseSeriesBlocks) (BootstrapResult, error) {
 	s.Lock()
-	defer s.Unlock()
+	defer s.bootstrapComplete()
 
+	var result BootstrapResult
 	if s.bs == bootstrapped {
-		return nil
+		return result, errSeriesAlreadyBootstrapped
 	}
-	if s.bs == bootstrapping {
-		return errSeriesIsBootstrapping
+
```

Comment:
why drop the `== bootstrapping` check?

Reply:
This state is getting removed, there's no such thing as bootstrapping, as we have a lock you're either bootstrapped or not bootstrapped.